mods/flowers/init.lua

```diff
@@ -204,38 +204,34 @@ minetest.register_node("flowers:mushroom_brown", {
 
 -- Mushroom spread and death
 
+function flowers.mushroom_spread(pos, node)
+	if minetest.get_node_light(pos, nil) == 15 then
+		minetest.remove_node(pos)
+		return
+	end
+	local positions = minetest.find_nodes_in_area_under_air(
+		{x = pos.x - 1, y = pos.y - 2, z = pos.z - 1},
+		{x = pos.x + 1, y = pos.y + 1, z = pos.z + 1},
```

Comment:
I'm not sure about the pos.y - 2 though, i know it makes sense for spread dowmwards through soil to an iair gap under soil, but that's an unlikely environment and the distance seems inconsistent.